### ISSUE TITLE
Support `"type": "module"` in package.json

### ISSUE THREAD

**Issue by lydell** [mpizenberg/elm-test-rs#68]
Steps to reproduce:

```
❯ echo '{ "private": true, "type": "module" }' > package.json

❯ elm init

❯ elm-test-rs init
The file test/Tests.elm was created

❯ elm-test-rs

elm-test-rs 1.0.0-alpha for elm 0.19.1
-----------------

Total time spent in the elm compiler: 0.80477065s
file:///Users/lydell/stuff/elm-test-rs-module/elm-stuff/tests-0.19.1/js/node_supervisor.js:1
process.chdir(__dirname);
              ^

ReferenceError: __dirname is not defined
    at file:///Users/lydell/stuff/elm-test-rs-module/elm-stuff/tests-0.19.1/js/node_supervisor.js:1:15
    at ModuleJob.run (internal/modules/esm/module_job.js:152:23)
    at async Loader.import (internal/modules/esm/loader.js:166:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)
```

https://nodejs.org/api/packages.html#packages_package_json_and_file_extensions

> Within a package, the package.json "type" field defines how Node.js should interpret .js files. If a package.json file does not have a "type" field, .js files are treated as CommonJS.
>
> A package.json "type" value of "module" tells Node.js to interpret .js files within that package as using ES module syntax.

- node-test-runner fix: https://github.com/rtfeldman/node-test-runner/pull/494
- node-elm-review fix: https://github.com/jfmengels/node-elm-review/commit/f5807ae0257443fb29c382bf34be9e8e4349b2f9
- See also: https://discord.com/channels/534524278847045633/729776405524512789/809498121544400908 and https://discord.com/channels/534524278847045633/729776405524512789/809519176413610084

**Comment by mpizenberg** [comment]
oh I understand better thanks @lydell ! Could we simply use the `.cjs` extension on our JS files?

**Comment by lydell** [comment]
I’m unsure about the support for .cjs files, but I tried down to Node.js 8 and it seemed to work, so it should probably be fine.

Note that `elm make --output=foo.cjs` doesn’t work – Elm mandates .js. So you’d have to rename files if you use that.

So – either .cjs or generating a package.json should be fine I think!

**Comment by mpizenberg** [comment]
Indeed, it seems like the most straightforward solution! because .cjs is not easy to use with the compiler and module type is not propagated from the parent .cjs file.

**Comment by mpizenberg** [comment]
Thanks, fixed in 5b30ea3
